notebooks/02_model/fm_deep_dive.ipynb

```diff
@@ -0,0 +1,894 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "metadata": {
+    "tags": [
+     "parameters"
+    ]
+   },
+   "outputs": [],
+   "source": [
+    "# Parameters\n",
+    "YAML_FILE_NAME = \"xDeepFM.yaml\"\n",
+    "TRAIN_FILE_NAME = \"cretio_tiny_train\"\n",
+    "VALID_FILE_NAME = \"cretio_tiny_valid\"\n",
+    "TEST_FILE_NAME = \"cretio_tiny_test\"\n",
+    "MODEL_FILE_NAME = \"model.out\"\n",
+    "OUTPUT_FILE_NAME = \"output.txt\"\n",
+    "\n",
+    "LEARNING_RATE = 0.2\n",
+    "LAMBDA = 0.002\n",
+    "# The metrics for binary classification options are \"acc\", \"prec\", \"f1\" and \"auc\"\n",
+    "# for regression, options are \"rmse\", \"mae\", \"mape\"\n",
+    "METRIC = \"auc\" \n",
+    "EPOCH = 10\n",
+    "OPT_METHOD = \"sgd\" # options are \"sgd\", \"adagrad\" and \"ftrl\""
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stderr",
+     "output_type": "stream",
+     "text": [
+      "100%|██████████| 10.3k/10.3k [00:01<00:00, 8.67kKB/s]\n"
+     ]
+    }
+   ],
+   "source": [
+    "tmpdir = TemporaryDirectory()\n",
+    "\n",
+    "data_path = tmpdir.name\n",
+    "yaml_file = os.path.join(data_path, YAML_FILE_NAME)\n",
+    "train_file = os.path.join(data_path, TRAIN_FILE_NAME)\n",
+    "valid_file = os.path.join(data_path, VALID_FILE_NAME)\n",
+    "test_file = os.path.join(data_path, TEST_FILE_NAME)\n",
+    "model_file = os.path.join(data_path, MODEL_FILE_NAME)\n",
+    "output_file = os.path.join(data_path, OUTPUT_FILE_NAME)\n",
+    "\n",
+    "if not os.path.exists(yaml_file):\n",
+    "    download_deeprec_resources(r'https://recodatasets.blob.core.windows.net/deeprec/', data_path, 'xdeepfmresources.zip')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The following steps are from the [official documentation of `xlearn`](https://xlearn-doc.readthedocs.io/en/latest/index.html) for building a model. To begin with, we do not modify any training parameter values. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "NOTE, if `xlearn` is run through command line, the training process can be displayed in the console."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Training task\n",
+    "ffm_model = xl.create_ffm()        # Use field-aware factorization machine (ffm)\n",
+    "ffm_model.setTrain(train_file)     # Set the path of training dataset\n",
+    "ffm_model.setValidate(valid_file)  # Set the path of validation dataset\n",
+    "\n",
+    "# Parameters:\n",
+    "#  0. task: binary classification\n",
+    "#  1. learning rate: 0.2\n",
+    "#  2. regular lambda: 0.002\n",
+    "#  3. evaluation metric: auc\n",
+    "#  4. number of epochs: 10\n",
+    "#  5. optimization method: sgd\n",
+    "param = {\"task\":\"binary\", \n",
```

Comment:
Same here, about formatting... but again, not super important. No pressure. ;-)

Reply:
added issue to work on it when we have time https://github.com/microsoft/recommenders/issues/1000